packages/interface/src/utils/fetchProjects.ts

```diff
@@ -89,3 +91,47 @@ export async function fetchApprovedProjects(registryAddress: string): Promise<IR
 
   return recipients ?? [];
 }
+
+/**
+ * Fetch all approved projects with metadata
+ * @param registryAddress
+ * @returns the projects with metadata values
+ */
+export async function fetchApprovedProjectsWithMetadata(
+  search: string,
+  registryAddress: string,
+): Promise<(IRecipient | null)[]> {
+  const response = await cachedFetch<{ recipients: IRecipient[] }>(config.maciSubgraphUrl, {
+    method: "POST",
+    body: JSON.stringify({
+      query: ApprovedProjects,
+      variables: { registryAddress },
+    }),
+  }).then((resp: GraphQLResponse) => resp.data?.recipients);
+
+  if (!response) {
+    return [];
+  }
+
+  const recipients = await Promise.all(
+    response.map(async (request) => {
+      const metadata = (await fetchMetadata(request.metadataUrl)) as unknown as Application;
+      if (search !== "") {
+        const name = metadata.name.toLowerCase();
+        if (!name.includes(search.trim().toLowerCase())) {
+          return null;
+        }
+      }
+      return {
```

Comment:
And don't use nested ifs
```suggestion
      const metadata = (await fetchMetadata(request.metadataUrl)) as unknown as Application;
  
      if (search !== "") {
        const name = metadata.name.toLowerCase();
        if (!name.includes(search.trim().toLowerCase())) {
          return null;
        }
      }
     
      return {
```